gusto/linear_solvers.py

```diff
@@ -220,6 +220,182 @@ def solve(self):
         self.theta_solver.solve()
         theta.assign(self.theta)
 
+class HybridisedCompressibleSolver(TimesteppingSolver):
+    """
+    Timestepping linear solver object for the compressible equations
+    in theta-pi formulation with prognostic variables u,rho,theta.
+
+    This solver follows the following strategy:
+    (1) Analytically eliminate theta (introduces error near topography)
+    (2a) Solve resulting system for (u[broken],rho,lambda) using hybridised
+    solver
+    (2b) reconstruct unbroken u
+    (3) Reconstruct theta
+
+    :arg state: a :class:`.State` object containing everything else.
+    :arg quadrature degree: tuple (q_h, q_v) where q_h is the required
+    quadrature degree in the horizontal direction and q_v is that in
+    the vertical direction
+    :arg params (optional): solver parameters
+    """
+
+    def __init__(self, state, quadrature_degree=None, params=None):
+
+        self.state = state
+
+        if quadrature_degree is not None:
+            self.quadrature_degree = quadrature_degree
+        else:
+            dgspace = state.spaces("DG")
+            if any(deg > 2 for deg in dgspace.ufl_element().degree()):
+                warning("default quadrature degree most likely not sufficient for this degree element")
+            self.quadrature_degree = (5, 5)
+
+        if params is None:
+            self.params = {'pc_type': 'fieldsplit',
+                           'pc_fieldsplit_type': 'schur',
+                           'ksp_type': 'gmres',
+                           'ksp_max_it': 100,
+                           'ksp_gmres_restart': 50,
+                           'pc_fieldsplit_schur_fact_type': 'FULL',
+                           'pc_fieldsplit_schur_precondition': 'selfp',
+                           'fieldsplit_0_ksp_type': 'preonly',
+                           'fieldsplit_0_pc_type': 'bjacobi',
+                           'fieldsplit_0_sub_pc_type': 'ilu',
+                           'fieldsplit_1_ksp_type': 'preonly',
+                           'fieldsplit_1_pc_type': 'gamg',
+                           'fieldsplit_1_mg_levels_ksp_type': 'chebyshev',
+                           'fieldsplit_1_mg_levels_ksp_chebyshev_estimate_eigenvalues': True,
+                           'fieldsplit_1_mg_levels_ksp_chebyshev_estimate_eigenvalues_random': True,
```

Comment:
Yes, these defaults are just copy pasted and haven't been set for this class officially yet